tools/svsm/SVSM_setup.sh

```diff
@@ -0,0 +1,486 @@
+#!/bin/bash
+
+set -eux
+set -eE
+set -u
+set -o pipefail
+
+# Function to display usage information
+usage() {
+    cat <<EOF
+Usage: $0 {setup_install|build_host_kernel|build_guest_kernel|install_igvm|build_qemu|build_edk2|build_svsm|build_all}
+
+Commands:
+  setup_install      Install all necessary dependencies and tools
+  build_host_kernel  Build the host kernel with SVSM support
+  build_guest_kernel Build the guest kernel with SVSM support
+  install_igvm       Install the IGVM tool
+  build_qemu         Build QEMU with SVSM support
+  build_edk2         Build EDK2 with SVSM support
+  build_svsm         Build coconut-svsm
+  build_all          Perform all the above actions in sequence
+EOF
+}
+
+# Working directory setup
+WORKING_DIR="${WORKING_DIR:-$HOME/coconut-svsm}"
+SETUP_WORKING_DIR="${SETUP_WORKING_DIR:-${WORKING_DIR}/setup}"
+
+# Check if the directory exists
+if [ -d "$WORKING_DIR" ]; then
+  echo "Directory '$WORKING_DIR' already exists and is not empty. Deleting it..."
+  rm -rf "WORKIG_DIR/"
+  WORKING_DIR="${WORKING_DIR:-$HOME/coconut-svsm}"
+fi
+
+export WORKSPACE=${WORKSPACE:-"/home/amd"}
+export PKG_CONFIG_PATH=${PKG_CONFIG_PATH:-""}
+
+
+
+# Function to log messages
+log() {
+    local level="$1"
+    local message="$2"
+    echo "[$(date +'%Y-%m-%d %H:%M:%S')] $level: $message"
+}
+
+# Function to check command success
+check_command() {
+    local command="$1"
+    local description="$2"
+    if eval "$command"; then
+        log "INFO" "$description: SUCCESS"
+    else
+        log "ERROR" "$description: FAILED"
+        exit 1
+    fi
+}
+
+# Function to install dependencies
+install_dependencies() {
+    log "INFO" "Updating package list..."
+    check_command "sudo apt update" "Failed to update package list"
+
+    log "INFO" "Installing dependencies..."
+    check_command "sudo apt install -y \
+        build-essential \
+        curl \
+        git \
+        libclang-dev \
+        autoconf \
+        autoconf-archive \
+        pkg-config \
+        automake \
+        libssl-dev \
+        perl \
+        libc6-dev \
+        gcc-multilib \
+        make \
+        gcc \
+        binutils" "Failed to install dependencies"
+}
+
+# Function to install Rust
+install_rust() {
```

Comment:
Either in here, or in the `install_all()` function we should probably check if they already have rust installed.

Reply:
Agreed. The updated SVSM_setup.sh script calls the install_rust function inside the install_dependencies function, even before the OS detection function is invoked.
 This is a one-time call during the host setup.